### PR TITLE
Updated mysql to 8.0.1, added ability to do package_version and set d…

### PR DESCRIPTION
**Pull request by rleibman**
Updated mysql to 8.0.1, added ability to do package_version and set default gem_version to 0.4.4

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@
 supports 'oracle'
 
 depends 'build-essential'
-depends 'mysql', '>= 6.0'
+depends 'mysql',               '~> 8.0.1'
```
Comment (jeffbyrnes) [inline metadata.rb]: This is a very tight pin; it’d be better to set this to `~> 8.0` to allow all non-breaking changes (i.e., what happens when mysql v8.1.0 ships?).

**Comment by jeffbyrnes** [comment]
Just ran into a situation where the lack of being able to specify a package_version has bitten me, since apparently the version in the `mysql` cookbook is incorrect for Ubuntu 16.04.

Would love to see this merged in!

File: metadata.rb (hunk 1 above)
Comment (klyonrad) [inline metadata.rb]: and did you test if it's actually necessary to use the Cookbook version above Version 8?

File: libraries/resource_mysql2_chef_gem.rb [hunk 2]
```diff
@@ -8,8 +8,9 @@ class Mysql2ChefGem < Chef::Resource::LWRPBase
       default_action :install
 
       attribute :mysql2_chef_gem_name, kind_of: String, name_attribute: true, required: true
-      attribute :gem_version, kind_of: String, default: '0.3.17'
+      attribute :gem_version, kind_of: String, default: '0.4.4'
       attribute :client_version, kind_of: String, default: nil
+      attribute :package_version, kind_of: String, default: nil
```
Comment (jeffbyrnes) [inline libraries/resource_mysql2_chef_gem.rb]: The `gem_version` property refers to the version of the `mysql2` gem that is installed, `package_version` refers to the APT/yum version of the `mysql-client` package.

Please see the [`mysql_client` resource](https://github.com/chef-cookbooks/mysql/blob/v8.0.4/libraries/mysql_client_installation_package.rb#L17-L23).

Comment (scassiba) [inline libraries/resource_mysql2_chef_gem.rb]: Awesome. Thanks for the clarity. There should be corresponding README changes to reflect the new functionality.

File: metadata.rb (hunk 1 above)
Comment (tas50) [inline metadata.rb]: This pin should be ~> 8.0

File: metadata.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@
 license 'Apache 2.0'
 description 'Provides the mysql2_chef_gem resource'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version '1.1.0'
+version '1.2.0'
```
Comment (tas50) [inline metadata.rb]: In general you shouldn't bump the version of a cookbook in a PR. That's usually done during release time

File: libraries/resource_mysql2_chef_gem.rb (hunk 2 above)
Comment (tas50) [inline libraries/resource_mysql2_chef_gem.rb]: Let's get this new property into the readme

**Comment by tas50** [comment]
The package version logic in the mysql cookbook has been reverted so this PR should no longer be necessary. Update to the latest version of mysql and it should all just work.

**Pull request closed without merging**